### PR TITLE
Correct the noise calculation in simultaneous inpainting function

### PR DESCRIPTION
**Pull request by tyler-a-cox**

### CODE REVIEW

File: hera_cal/lst_stack/averaging.py [hunk 1]
```diff
@@ -657,16 +702,25 @@ def average_and_inpaint_simultaneously(
         stack, get_std=False, get_mad=False, inpainted_mode=False, mean_fill_value=0.0
     )
 
+    # Map polarizations to pol indices for correct noise variance computation
+    pol_to_idx = {}
```
Comment (jsdillon) [inline hera_cal/lst_stack/averaging.py:706]: I'd clarify that this is mapping antenna polarizations (antpol / jpol) to visibility pol idx.

File: hera_cal/lst_stack/averaging.py [hunk 2]
```diff
@@ -681,11 +735,15 @@ def average_and_inpaint_simultaneously(
             stackd = stack.data[:, iap, :, polidx]
             stackf = complete_flags[:, iap, :, polidx]
             stackn = np.abs(stack.nsamples[:, iap, :, polidx])
-            flagged_mean = lstavg['data'][iap, :, polidx]
-            avg_flgs = lstavg['flags'][iap, :, polidx]
+            flagged_mean = lstavg["data"][iap, :, polidx]
+            avg_flgs = lstavg["flags"][iap, :, polidx]
 
+            pol1, pol2 = utils.split_pol(pol)
```
Comment (jsdillon) [inline hera_cal/lst_stack/averaging.py:741]: same note as before, I'd call these `jpol1` or `antpol1` or `ap1` or `ant_1_pol` or something like that.

**Review by jsdillon — CHANGES_REQUESTED** [review]
Looks basically fine. Just a few suggested chanages to make the code cleaner and more readable.

File: hera_cal/lst_stack/averaging.py [hunk 3]
```diff
@@ -326,25 +328,31 @@ def reduce_lst_bins(
...
         # We have no data to reduce (no data in this LST bin)
         return {
-            'data': np.nan * np.ones(data.shape[1:], dtype=complex),
-            'flags': np.ones(data.shape[1:], dtype=bool),
-            'std': np.inf * np.ones(data.shape[1:], dtype=complex),
-            'nsamples': np.zeros(data.shape[1:]),
-            'days_binned': np.zeros(data.shape[1:]),
-            'median': np.nan * np.ones(data.shape[1:], dtype=complex) if get_mad else None,
-            'mad': np.inf * np.ones(data.shape[1:], dtype=complex) if get_mad else None,
+            "data": np.nan * np.ones(data.shape[1:], dtype=complex),
```
[the remark this answers is not in the document]
Comment (tyler-a-cox) [reply hera_cal/lst_stack/averaging.py]: Fixed

File: hera_cal/lst_stack/averaging.py (hunk 1 above)
Comment (tyler-a-cox) [reply hera_cal/lst_stack/averaging.py:704]: Done

**Review by jsdillon — APPROVED** [review]
Assuming tests pass, I'm good to go with this.

**Pull request merged**